.github/workflows/e2e-test-playwright.yml

```diff
@@ -9,12 +9,26 @@ on:
 jobs:
   tests-ui-e2e:
     name: UI end-to-end tests
-    if: github.repository == 'meshery/meshery'
+    if: >
+      github.repository == 'meshery/meshery' &&
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
     runs-on: ubuntu-22.04
-    permissions:
-      issues: write
-      pull-requests: write
     steps:
+      - name: Download meshery-ui artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: meshery-ui
+          path: /home/runner/work/meshery/meshery/ui/out
+          github-token: ${{ github.GH_ACCESS_TOKEN }}
+          run-id: ${{ github.event.workflow_run.id }}
```

Comment:
Let's see if `github.event.workflow_run.id` the right ID

Reply:
Good point, @jerensl.